pkg/specgen/generate/storage.go

```diff
@@ -314,8 +314,8 @@ func addContainerInitBinary(s *specgen.SpecGenerator, path string) (spec.Mount,
 	if !s.PidNS.IsPrivate() {
 		return mount, fmt.Errorf("cannot add init binary as PID 1 (PID namespace isn't private)")
 	}
-	if s.Systemd == "true" || s.Systemd == "always" {
-		return mount, fmt.Errorf("cannot use container-init binary with systemd")
+	if s.Systemd == "always" {
```

Comment:
Is this change right?  Can we used the container-init binary with systemd?

Reply:
`--systemd=true` is the default from the CLI, which implies that we use systemd if requested; I think we need to ignore it when both it and `--init` are given, in favor of the more-specific `--init`. Meanwhile, `--systemd=always` forces systemd on, and that sounds like it should conflict.